src/main.rs

```diff
@@ -142,24 +143,37 @@ async fn main() {
                 })
                 .buffer_unordered(1);
 
-            let q = translation_result
+            // Take each key-value pair from translation_result and insert it into the output hashmap
+            // while maintaining a bucketed count of translation quality scores
+            let (q, good_count, okay_count, poor_count) = translation_result
                 .fold(
-                    destination_hash_map,
-                    |mut destination_hash_map, (key, value)| async {
+                    (destination_hash_map, 0, 0, 0),
+                    |(mut destination_hash_map, mut good_count, mut okay_count, mut poor_count),
+                    (key, value)| async move {
                         let translation = match value {
                             Ok(translation) => translation,
                             Err(error) => {
                                 println!("Error translating string: {}", error);
-                                return destination_hash_map;
+                                return (destination_hash_map, good_count, okay_count, poor_count);
                             }
                         };
+
+                        if translation.quality_score > 0.6 {
```

Comment:
One little bit but can we move the cutoff values to constants somewhere. I have a hunch they will change in the near term and want to make it easier to track down laterZ